Tests/SubprocessTests/SubprocessTests+Unix.swift

```diff
@@ -929,10 +953,124 @@ extension SubprocessUnixTests {
+            [0xE2, 0x80, 0xA9]  // Paragraph separator
+        ]
+
+        // Generate test cases
+        func generateString(size: TestCaseSize) -> [UInt8] {
+            // Basic Latin has the range U+0020 ... U+007E
+            let range: ClosedRange<UInt8> = 0x20 ... 0x7E // 0x4E ... 0x5A
+
+            let length: Int
+            switch size {
+            case .large:
+                length = Int(Double.random(in: 1.0 ..< 2.0) * Double(readBufferSize))
+            case .medium:
+                length = Int(Double.random(in: 0.2 ..< 1.0) * Double(readBufferSize))
+            case .small:
+                length = Int.random(in: 0 ..< 16)
+            }
+
+            var buffer: [UInt8] = Array(repeating: 0, count: length)
+            for index in 0 ..< length {
+                buffer[index] = UInt8.random(in: range)
+            }
+            return buffer
+        }
+
+        // Generate at least 2 long lines that is longer than buffer size
+        func generateTestCases(count: Int) -> [TestCase] {
+            var targetSizes: [TestCaseSize] = TestCaseSize.allCases.flatMap {
+                Array(repeating: $0, count: count / 3)
+            }
+            // Fill the remainder
+            let remaining = count - targetSizes.count
+            let rest = TestCaseSize.allCases.shuffled().prefix(remaining)
+            targetSizes.append(contentsOf: rest)
+            // Do a final shuffle to achieve random order
+            targetSizes.shuffle()
+            // Now generate test cases based on sizes
+            var testCases: [TestCase] = []
+            for size in targetSizes {
+                let components = generateString(size: size)
+                // Choose a random new line
+                let newLine = newLineCharacters.randomElement()!
+                let string = String(decoding: components + newLine, as: UTF8.self)
+                testCases.append((
+                    value: string,
+                    count: components.count + newLine.count,
+                    newLine: String(decoding: newLine, as: UTF8.self)
+                ))
+            }
+            return testCases
+        }
+
+        func writeTestCasesToFile(_ testCases: [TestCase], at url: URL) throws {
+            #if canImport(Darwin)
+            FileManager.default.createFile(atPath: url.path(), contents: nil, attributes: nil)
+            let fileHadle = try FileHandle(forWritingTo: url)
+            for testCase in testCases {
+                fileHadle.write(testCase.value.data(using: .utf8)!)
+            }
+            try fileHadle.close()
+            #else
+            var result = ""
+            for testCase in testCases {
+                result += testCase.value
+            }
+            try result.write(to: url, atomically: true, encoding: .utf8)
+            #endif
+        }
+
+        let testCaseCount = 60
+        let testFilePath = URL.temporaryDirectory.appending(path: "NewLines.txt")
```

Comment:
Throw a `UUID` in here, or test cases that are running in parallel will reuse the same file.